src/Microsoft.ML.Data/StaticPipe/Estimator.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Data.StaticPipe.Runtime;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+
+namespace Microsoft.ML.Data.StaticPipe
+{
+    public sealed class Estimator<TTupleInShape, TTupleOutShape, TTransformer> : SchemaBearing<TTupleOutShape>
+        where TTransformer : class, ITransformer
+    {
+        public IEstimator<TTransformer> AsDynamic { get; }
+        private readonly StaticSchemaShape _inShape;
+
+        internal Estimator(IHostEnvironment env, IEstimator<TTransformer> estimator, StaticSchemaShape inShape, StaticSchemaShape outShape)
```

Comment:
Can we give a better name for the `estimator` parameter?  It is confusing that this class is named `Estimator`, and it takes an `estimator` in its constructor.  Is that the `innerEstimator`?  `baseEstimator`?  `previousEstimator`?  etc.